cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp

```diff
@@ -20,18 +20,52 @@
  */
 #include "exec/nextgen/jitlib/llvmjit/LLVMJITModule.h"
 
-#include <llvm/IR/Function.h>
+#include <llvm/Bitcode/BitcodeReader.h>
+#include <llvm/IR/Module.h>
+#include <llvm/Transforms/Utils/Cloning.h>
+
+#include <memory>
+#include <mutex>
 
 #include "exec/nextgen/jitlib/llvmjit/LLVMJITUtils.h"
 #include "util/Logger.h"
+#include "util/filesystem/cider_path.h"
 
 namespace cider::jitlib {
 
-LLVMJITModule::LLVMJITModule(const std::string& name)
+static llvm::MemoryBuffer* getExternalBuffer() {
+  static std::once_flag has_set_buffer;
+  static std::unique_ptr<llvm::MemoryBuffer> external_function_buffer;
+
+  std::call_once(has_set_buffer, [&]() {
+    auto root_path = cider::get_root_abs_path();
+    auto template_path = root_path + "/function/RuntimeFunctions.bc";
+    CHECK(boost::filesystem::exists(template_path));
+
+    auto buffer_or_error = llvm::MemoryBuffer::getFile(template_path);
+    CHECK(!buffer_or_error.getError()) << "bc_filename=" << template_path;
+    external_function_buffer = std::move(buffer_or_error.get());
+  });
+
+  return external_function_buffer.get();
+}
+
+LLVMJITModule::LLVMJITModule(const std::string& name, bool has_external_module)
     : context_(std::make_unique<llvm::LLVMContext>())
-    , module_(std::make_unique<llvm::Module>(name, *context_))
-    , engine_(nullptr) {
-  CHECK(context_);
+    , engine_(nullptr)
+    , has_external_module_(has_external_module) {
+  if (has_external_module_) {
```

Comment:
`has_external_module_` no need as a member? and rename to `should_copy_runtime_module` may be better.

Reply:
get it